Pilot Data Exploration.R

```diff
@@ -24,46 +24,213 @@ if(pull_fresh_dataset) {
+         & !is.na(pi_pfs_5) & !is.na(pi_pfs_6) & !is.na(pi_pfs_7))
+  #n = 81 eligible, Program Feedback Scale completers
+
+#Identify Pre-to-Post measures completers
+b_pi_completers <- eligible_responders %>%
+  filter(!is.na(b_bhs_sum) & !is.na(b_shs_agency_sum) & !is.na(b_fas_sum) & !is.na(b_bsss_sum) 
+         & !is.na(pi_bhs_sum) & !is.na(pi_shs_agency_sum) & !is.na(pi_fas_sum) & !is.na(pi_bsss_sum))
+  #n = 75 pre-to-post measure completers
 
 
+####  Analysis  ####
 
-####  Data Cleaning  ####
-## Create summary variables
-clean_data <- qualtrics_data %>%
-  mutate(b_fas_sum = b_fas_1 + b_fas_2 + b_fas_3 + b_fas_4 + b_fas_5 + b_fas_6 + b_fas_7,
-         pi_fas_sum = pi_fas_1 + pi_fas_2 + pi_fas_3 + pi_fas_4 + pi_fas_5 + pi_fas_6 + pi_fas_7)
+##Duration
+
+names_in_order <- qualtrics_data %>%
+  select(ends_with("Page Submit")) %>%
+  names()
+
+#Put mean duration for each page into a new data frame
+page_durations <- qualtrics_data %>% #change to b_pi_completers dataset? If I do, I get a NaN result for line 120 and I'm not sure why :/
+  select(ends_with("Page Submit")) %>%
+  pivot_longer(cols = everything(),
+               names_to = "page",
+               values_to = "duration") %>%
+  group_by(page) %>%
+  summarise(mean_duration = mean(duration, na.rm = T),
+            sd_duration = sd(duration,  na.rm = T)) #seems odd/incorrect that the values I'm getting for standard deviation are so giant
+
+page_durations$page <- factor(page_durations$page, levels = names_in_order)
+
+#Plot mean duration for each page
+ggplot(data = page_durations) +
+  geom_col(aes(x = page, y = mean_duration)) +
+  coord_flip()
+
+#Identify mean duration for intro and pre-SSI questionnaires, the SSI, and post-SSI questionnaires
+b_ssi_pi_mean_durations <- page_durations %>% 
+  pivot_wider(names_from = page, values_from = mean_duration) %>% 
+  rename_with(~str_replace(.,'tim_Page Submit', 'page_submit')) %>% #didn't need this but leaving for my future reference
+  mutate(b_mean_duration_sec = sum(across(starts_with("b_")))) %>%  
+  mutate(b_mean_duration_minute = b_mean_duration_sec / 60) %>% 
+  mutate(ssi_mean_duration_sec = sum(across(starts_with("bn_")))) %>% #how to account for the fact that not every single person saw every page (e.g., the multiple story options)?
+  mutate(ssi_mean_duration_minute = ssi_mean_duration_sec / 60) %>%
+  mutate(pi_mean_duration_sec = sum(across(starts_with("pi_")))) %>% 
+  mutate(pi_mean_duration_minute = pi_mean_duration_sec / 60)
+
+b_ssi_pi_mean_durations$b_mean_duration_minute #4.112764 minutes
+b_ssi_pi_mean_durations$ssi_mean_duration_minute #27.76778 minutes
+b_ssi_pi_mean_durations$pi_mean_duration_minute #3.57133 minutes
+                        
+#should we also do standard deviations? If so, what's the best way? 
+
+##Program Feedback Scale
+
+#Total score (items summed) - how to better explain the difference between this and the one below?
```

Comment:
I'd call this the mean and sd of each person's sum score. The mean and sum scores are equivalent, i.e. the sum score is just the mean score * 7

Reply:
So I don't think you need to calculate both

Reply:
Oh ok that makes sense! And the output of mean(pfs_completers$pi_pfs_sum) / 7 is totally the same as mean(as.matrix(pi_pfs_all_items)) – I was overthinking this one